test/language/expressions/arrow-function/arrow/apply_3b.js

```diff
@@ -0,0 +1,21 @@
+// Copyright 2015 Microsoft Corporation. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
+
+/*---
+description: invoke arrow function using apply - 'this' is lexically bound
+es6id: 14.2
+---*/
+
+var b = {
+method1: (arg) => {this.x = arg},
+method2: (arg) => {return (arg) => {this.y = arg}},
+method3: function(arg) {return (arg) => {this.z = arg}}
+}
+
+b.method1.apply(undefined, ["arg1"]);
+b.method2().apply(undefined, ["arg2"]);
+b.method3().apply(undefined, ["arg3"]);
+
+assert.sameValue(this.x, "arg1", "this binding was incorrect.");
+assert.sameValue(this.y, "arg2", "this binding was incorrect.");

```

Comment:
Duplicates

- https://github.com/tc39/test262/blob/master/test/language/expressions/arrow-function/lexical-this.js